scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -493,14 +494,18 @@ private[spel] class Typer(
         }
 
       case e: VariableReference =>
-        // only sane way of getting variable name :|
-        val name = e.toStringAST.substring(1)
-        validationContext
-          .get(name)
-          .orElse(current.stackHead.filter(_ => name == "this"))
-          .map(valid)
-          .getOrElse(invalid(UnresolvedReferenceError(name)))
-          .map(toNodeResult)
+        if (!absentVariableReferenceAllowed) {
+          // only sane way of getting variable name :|
+          val name = e.toStringAST.substring(1)
+          validationContext
```

Comment:
the code do sth different than variable says :) What about global variables and `this`? It should still work. Only invalid case should be handled differently. Please add tests for that